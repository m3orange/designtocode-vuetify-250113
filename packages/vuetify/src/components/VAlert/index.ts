export { default as VAlert } from './VAlert'
export { default as VAlertText } from './VAlertText'
