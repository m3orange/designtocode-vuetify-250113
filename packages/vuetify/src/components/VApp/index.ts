// Styles
import './VApp.sass'
import VApp from './VApp'

export { VApp }
export default VApp
