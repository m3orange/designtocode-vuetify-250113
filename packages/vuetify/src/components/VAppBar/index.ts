export { default as VAppBar } from './VAppBar'
