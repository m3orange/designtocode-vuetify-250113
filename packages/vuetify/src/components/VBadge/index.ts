// Styles
import './VBadge.sass'

import VBadge from './VBadge'

export { VBadge }
export default VBadge
