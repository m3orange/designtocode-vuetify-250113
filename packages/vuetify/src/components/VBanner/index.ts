// Styles
import './VBanner.sass'

import VBanner from './VBanner'

export { VBanner }
export default VBanner
