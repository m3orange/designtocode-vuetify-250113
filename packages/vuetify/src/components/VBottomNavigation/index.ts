// Styles
import './VBottomNavigation.sass'

import VBottomNavigation from './VBottomNavigation'

export { VBottomNavigation }
export default VBottomNavigation
