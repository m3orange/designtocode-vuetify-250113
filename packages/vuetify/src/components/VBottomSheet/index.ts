import './VBottomSheet.sass'

import VBottomSheet from './VBottomSheet'

export { VBottomSheet }
export default VBottomSheet
