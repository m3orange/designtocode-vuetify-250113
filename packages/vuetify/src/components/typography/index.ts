import {
  createSimpleFunctional
} from '../../util/helpers'

const VDisplay4 = createSimpleFunctional('display-4', 'div', 'v-display-4')
const VDisplay3 = createSimpleFunctional('display-3', 'div', 'v-display-3')
const VDisplay2 = createSimpleFunctional('display-2', 'div', 'v-display-2')
const VDisplay1 = createSimpleFunctional('display-1', 'div', 'v-display-1')
const VHeadline = createSimpleFunctional('headline', 'div', 'v-headline')
const VTitle = createSimpleFunctional('title', 'div', 'v-title')
const VSubtitle1 = createSimpleFunctional('subtitle-1', 'div', 'v-subtitle-1')
const VSubtitle2 = createSimpleFunctional('subtitle-2', 'div', 'v-subtitle-2')
const VBody1 = createSimpleFunctional('body-1', 'div', 'v-body-1')
const VBody2 = createSimpleFunctional('body-2', 'div', 'v-body-2')
const VCaption = createSimpleFunctional('caption', 'div', 'v-caption')
const VOverline = createSimpleFunctional('overline', 'div', 'v-overline')

export {
  VDisplay4,
  VDisplay3,
  VDisplay2,
  VDisplay1,
  VHeadline,
  VTitle,
  VSubtitle1,
  VSubtitle2,
  VBody1,
  VBody2,
  VCaption,
  VOverline
}

export default {
  $_vuetify_subcomponents: {
    VDisplay4,
    VDisplay3,
    VDisplay2,
    VDisplay1,
    VHeadline,
    VTitle,
    VSubtitle1,
    VSubtitle2,
    VBody1,
    VBody2,
    VCaption,
    VOverline
  }
}
