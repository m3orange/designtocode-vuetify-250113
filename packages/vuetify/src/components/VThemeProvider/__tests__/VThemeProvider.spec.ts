// Components
import { VThemeProvider } from '..'
import { VBtn } from '@/components'

// Utilities
import { h } from 'vue'
import { mount } from '@vue/test-utils'
import { createVuetify } from '@/framework'

describe('VThemeProvider.ts', () => {
  function mountFunction (options = {}, vuetifyOptions = {}) {
    const vuetify = createVuetify(vuetifyOptions)
    return mount(VThemeProvider, {
      global: {
        plugins: [vuetify],
      },
      ...options,
    })
  }

  it('should use theme defined in prop', async () => {
    const wrapper = mountFunction({
      props: { theme: 'dark' },
      slots: {
        default: () => h(VBtn, () => 'foo')
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should use default theme from options', async () => {
    const wrapper = mountFunction({
      slots: {
        default: () => h(VBtn, () => 'foo')
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render div when using with-background prop', () => {
    const wrapper = mountFunction({
      props: {
        withBackground: true,
      },
      slots: {
        default: () => h(VBtn, () => 'foo')
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
