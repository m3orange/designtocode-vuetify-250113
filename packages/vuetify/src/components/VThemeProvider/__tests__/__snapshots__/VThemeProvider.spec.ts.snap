// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VThemeProvider.ts should work match snapshot 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;
